assets/js/modules/analytics-4/utils/__fixtures__/mocked-pivot-report-multi-dimension-fieldnames.json

```diff
@@ -0,0 +1,437 @@
+        "dimensionValues": [
+          {
+            "value": "RESERVED_MIN"
+          },
+          {
+            "value": "city"
+          }
+        ],
+        "metricValues": [
+          {
+            "value": "27"
+          },
+          {
+            "value": "91"
+          }
+        ]
+      },
+      {
+        "dimensionValues": [
+          {
+            "value": "RESERVED_MIN"
+          },
+          {
+            "value": "audienceResourceName"
+          }
+        ],
+        "metricValues": [
+          {
+            "value": "86"
+          },
+          {
+            "value": "53"
+          }
+        ]
+      },
+      {
+        "dimensionValues": [
+          {
+            "value": "RESERVED_MIN"
+          },
+          {
+            "value": "country"
+          }
+        ],
+        "metricValues": [
+          {
+            "value": "38"
+          },
+          {
+            "value": "71"
+          }
+        ]
+      },
+      {
+        "dimensionValues": [
+          {
+            "value": "RESERVED_MAX"
+          },
+          {
+            "value": "city"
+          }
+        ],
+        "metricValues": [
+          {
+            "value": "21"
+          },
+          {
+            "value": "74"
+          }
+        ]
+      },
+      {
+        "dimensionValues": [
+          {
+            "value": "RESERVED_MAX"
+          },
+          {
+            "value": "audienceResourceName"
+          }
+        ],
+        "metricValues": [
+          {
+            "value": "66"
+          },
+          {
+            "value": "80"
+          }
+        ]
+      },
+      {
+        "dimensionValues": [
+          {
+            "value": "RESERVED_MAX"
+          },
+          {
+            "value": "country"
+          }
+        ],
+        "metricValues": [
+          {
+            "value": "16"
+          },
+          {
+            "value": "22"
+          }
+        ]
+      },
+      {
+        "dimensionValues": [
+          {
+            "value": "RESERVED_TOTAL"
+          },
+          {
+            "value": "city"
+          }
+        ],
+        "metricValues": [
+          {
+            "value": "45"
+          },
+          {
+            "value": "34"
+          }
+        ]
+      },
+      {
+        "dimensionValues": [
+          {
+            "value": "RESERVED_TOTAL"
+          },
+          {
+            "value": "audienceResourceName"
+          }
+        ],
+        "metricValues": [
+          {
+            "value": "23"
+          },
+          {
+            "value": "91"
+          }
+        ]
+      },
+      {
+        "dimensionValues": [
+          {
+            "value": "RESERVED_TOTAL"
+          },
+          {
+            "value": "country"
+          }
+        ],
+        "metricValues": [
+          {
+            "value": "38"
+          },
+          {
+            "value": "66"
+          }
+        ]
+      }
+    ],
+    "rows": [
```

Comment:
Lastly, the number of `rows` is not as expected either.

The number of rows should be the product of multiplying the row count for each pivot together. So with pivots returning 2, 3 and 3 rows, the report row count should be 2x3x3 = 18.

However the current output only has 6 rows, so it evidently doesn't have the right combination of values.

<details>
<summary>Expected shape (taken from my example)</summary>

```json
"rows": [
	{
		"dimensionValues": [
			{
				"value": "new"
			},
			{
				"value": "Dublin"
			},
			{
				"value": "Ireland"
			},
			{
				"value": "iOS"
			}
		],
		"metricValues": [
			{
				"value": "551"
			},
			{
				"value": "551"
			}
		]
	},
	{
		"dimensionValues": [
			{
				"value": "new"
			},
			{
				"value": "Dublin"
			},
			{
				"value": "Ireland"
			},
			{
				"value": "Android"
			}
		],
		"metricValues": [
			{
				"value": "430"
			},
			{
				"value": "430"
			}
		]
	},
	{
		"dimensionValues": [
			{
				"value": "new"
			},
			{
				"value": "Cork"
			},
			{
				"value": "Ireland"
			},
			{
				"value": "iOS"
			}
		],
		"metricValues": [
			{
				"value": "191"
			},
			{
				"value": "191"
			}
		]
	},
	{
		"dimensionValues": [
			{
				"value": "new"
			},
			{
				"value": "Dublin"
			},
			{
				"value": "Ireland"
			},
			{
				"value": "Windows"
			}
		],
		"metricValues": [
			{
				"value": "121"
			},
			{
				"value": "121"
			}
		]
	},
	{
		"dimensionValues": [
			{
				"value": "new"
			},
			{
				"value": "Sao Paulo"
			},
			{
				"value": "Brazil"
			},
			{
				"value": "iOS"
			}
		],
		"metricValues": [
			{
				"value": "83"
			},
			{
				"value": "83"
			}
		]
	},
	{
		"dimensionValues": [
			{
				"value": "returning"
			},
			{
				"value": "Dublin"
			},
			{
				"value": "Ireland"
			},
			{
				"value": "Android"
			}
		],
		"metricValues": [
			{
				"value": "78"
			},
			{
				"value": "78"
			}
		]
	},
	{
		"dimensionValues": [
			{
				"value": "returning"
			},
			{
				"value": "Dublin"
			},
			{
				"value": "Ireland"
			},
			{
				"value": "iOS"
			}
		],
		"metricValues": [
			{
				"value": "70"
			},
			{
				"value": "70"
			}
		]
	},
	{
		"dimensionValues": [
			{
				"value": "new"
			},
			{
				"value": "Sao Paulo"
			},
			{
				"value": "Brazil"
			},
			{
				"value": "Android"
			}
		],
		"metricValues": [
			{
				"value": "64"
			},
			{
				"value": "64"
			}
		]
	},
	{
		"dimensionValues": [
			{
				"value": "new"
			},
			{
				"value": "Sao Paulo"
			},
			{
				"value": "Brazil"
			},
			{
				"value": "Windows"
			}
		],
		"metricValues": [
			{
				"value": "36"
			},
			{
				"value": "36"
			}
		]
	},
	{
		"dimensionValues": [
			{
				"value": "returning"
			},
			{
				"value": "Cork"
			},
			{
				"value": "Ireland"
			},
			{
				"value": "iOS"
			}
		],
		"metricValues": [
			{
				"value": "26"
			},
			{
				"value": "26"
			}
		]
	},
	{
		"dimensionValues": [
			{
				"value": "new"
			},
			{
				"value": "Cork"
			},
			{
				"value": "Ireland"
			},
			{
				"value": "Android"
			}
		],
		"metricValues": [
			{
				"value": "21"
			},
			{
				"value": "21"
			}
		]
	},
	{
		"dimensionValues": [
			{
				"value": "returning"
			},
			{
				"value": "Dublin"
			},
			{
				"value": "Ireland"
			},
			{
				"value": "Windows"
			}
		],
		"metricValues": [
			{
				"value": "14"
			},
			{
				"value": "14"
			}
		]
	},
	{
		"dimensionValues": [
			{
				"value": "returning"
			},
			{
				"value": "Sao Paulo"
			},
			{
				"value": "Brazil"
			},
			{
				"value": "Android"
			}
		],
		"metricValues": [
			{
				"value": "12"
			},
			{
				"value": "12"
			}
		]
	},
	{
		"dimensionValues": [
			{
				"value": "new"
			},
			{
				"value": "Cork"
			},
			{
				"value": "Ireland"
			},
			{
				"value": "Windows"
			}
		],
		"metricValues": [
			{
				"value": "8"
			},
			{
				"value": "8"
			}
		]
	},
	{
		"dimensionValues": [
			{
				"value": "returning"
			},
			{
				"value": "Sao Paulo"
			},
			{
				"value": "Brazil"
			},
			{
				"value": "iOS"
			}
		],
		"metricValues": [
			{
				"value": "7"
			},
			{
				"value": "7"
			}
		]
	},
	{
		"dimensionValues": [
			{
				"value": "returning"
			},
			{
				"value": "Cork"
			},
			{
				"value": "Ireland"
			},
			{
				"value": "Android"
			}
		],
		"metricValues": [
			{
				"value": "4"
			},
			{
				"value": "4"
			}
		]
	},
	{
		"dimensionValues": [
			{
				"value": "returning"
			},
			{
				"value": "Sao Paulo"
			},
			{
				"value": "Brazil"
			},
			{
				"value": "Windows"
			}
		],
		"metricValues": [
			{
				"value": "4"
			},
			{
				"value": "4"
			}
		]
	},
	{
		"dimensionValues": [
			{
				"value": "returning"
			},
			{
				"value": "Cork"
			},
			{
				"value": "Ireland"
			},
			{
				"value": "Windows"
			}
		],
		"metricValues": [
			{
				"value": "1"
			},
			{
				"value": "1"
			}
		]
	}
],
```

<details>
<summary>Current output</summary>

```json
    "rows": [
      {
        "dimensionValues": [
          {
            "value": "properties/12345/audiences/2"
          },
          {
            "value": "Cork"
          },
          {
            "value": "India"
          },
          {
            "value": "India"
          }
        ],
        "metricValues": [
          {
            "value": "98"
          },
          {
            "value": "7"
          }
        ]
      },
      {
        "dimensionValues": [
          {
            "value": "properties/12345/audiences/2"
          },
          {
            "value": "(not set)"
          },
          {
            "value": "United Kingdom"
          },
          {
            "value": "United Kingdom"
          }
        ],
        "metricValues": [
          {
            "value": "89"
          },
          {
            "value": "62"
          }
        ]
      },
      {
        "dimensionValues": [
          {
            "value": "properties/12345/audiences/1"
          },
          {
            "value": "Dublin"
          },
          {
            "value": "United States"
          },
          {
            "value": "United States"
          }
        ],
        "metricValues": [
          {
            "value": "55"
          },
          {
            "value": "14"
          }
        ]
      },
      {
        "dimensionValues": [
          {
            "value": "properties/12345/audiences/1"
          },
          {
            "value": "Cork"
          },
          {
            "value": "India"
          },
          {
            "value": "India"
          }
        ],
        "metricValues": [
          {
            "value": "17"
          },
          {
            "value": "39"
          }
        ]
      },
      {
        "dimensionValues": [
          {
            "value": "properties/12345/audiences/2"
          },
          {
            "value": "Dublin"
          },
          {
            "value": "United States"
          },
          {
            "value": "United States"
          }
        ],
        "metricValues": [
          {
            "value": "10"
          },
          {
            "value": "73"
          }
        ]
      },
      {
        "dimensionValues": [
          {
            "value": "properties/12345/audiences/1"
          },
          {
            "value": "(not set)"
          },
          {
            "value": "United Kingdom"
          },
          {
            "value": "United Kingdom"
          }
        ],
        "metricValues": [
          {
            "value": "3"
          },
          {
            "value": "13"
          }
        ]
      }
    ],
```

</details>